src/components/modals/FormStepSaveModal.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Display a modal to allow the user to save the form step in it's current state.
+ */
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {useHistory} from 'react-router-dom';
+import {FormattedMessage} from 'react-intl';
+import {put, post, destroy} from 'api';
+import {getBEMClassName} from 'utils';
+import Modal from 'components/modals/Modal';
+import ErrorMessage from 'components/ErrorMessage';
+import Body from 'components/Body';
+import Label from 'components/Label';
+import Input from 'components/Input';
+import HelpText from 'components/HelpText';
+import {Toolbar, ToolbarList} from 'components/Toolbar';
+import Button from 'components/Button';
+
+
+const FormStepSaveModal = ({
+    isOpen,
+    closeModal,
+    stepData,
+    saveStepDataUrl,
+    suspendFormUrl,
+    destroySessionUrl,
+}) => {
```

Comment:
we have a very specific component here tightly coupled to the `FormStep` - please pass the `submission` and `submissionStep` props in their entirety rather than just some attributes. I think we can reduce the number of props for the component that way.

Reply:
Do we still want to do this?  As far as I can see the `stepData` needs to be passed in separately which means we would still have five props (I removed the `destroySessionUrl` already) so I think we would have 5 props either way.

Two of the reasons I kept them as just the url:
1. It feels like overkill to pass an entire object when we only want one property of that object
2. It does seem like this would make it harder to reuse this modal in a different component since it'd make it more tightly coupled to the `FormStep` modal.